### ISSUE TITLE
ES6 Template Strings and no-unused-vars aren't (competely) compatible

### ISSUE THREAD

**Issue by zpao** [eslint/eslint#1542]
```js
var who = "Paul";
module.exports = `Hello ${who}!`;
```

Linting that results in a warning that `who is defined but never used`. My human eyes can tell you that's wrong but I'm not actually sure if esprima provides enough info to also tell us that.

This is using v0.11.0-alpha.0 and config using ES6.

**Comment by nzakas** [comment]
Working on this.

**Comment by sandwichsudo** [comment]
Got this problem with v 1.10.3

**Comment by seemsindie** [comment]
How do you turn this on? :)

**Comment by KeitIG** [comment]
bump ! :)

**Comment by platinumazure** [comment]
@seemsindie @KeitIG Please stop by our [Gitter chat](https://gitter.im/eslint/eslint) if you need help with this. Thanks!
